Sources/_SwiftSyntaxTestSupport/FixItApplier.swift

```diff
@@ -0,0 +1,149 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftSyntax
+
+public enum FixItApplier {
+  fileprivate struct Edit: Equatable {
+    var startUtf8Offset: Int
+    var endUtf8Offset: Int
+    let replacement: String
+
+    var originalLength: Int {
+      return endUtf8Offset - startUtf8Offset
+    }
+
+    var replacementLength: Int {
+      return replacement.utf8.count
+    }
+
+    var replacementRange: Range<Int> {
+      return startUtf8Offset..<endUtf8Offset
+    }
+  }
+
+  /// Applies selected or all Fix-Its from the provided diagnostics to a given syntax tree.
+  ///
+  /// - Parameters:
+  ///   - diagnostics: An array of `Diagnostic` objects, each containing one or more Fix-Its.
+  ///   - filterByMessages: An optional array of message strings to filter which Fix-Its to apply.
+  ///                      If `nil`, all Fix-Its in `diagnostics` are applied.
```

Comment:
AFAICT this comment isn’t actually correct.

```suggestion
  ///     If `nil`, the first Fix-It from each diagnostic is applied.
```